TUTORIAL_USE.md

```diff
@@ -0,0 +1,118 @@
+# Using the existing code generators
+
+Java::Geci provides several out-of-the-box code generators. 
+In this tutorial we'll use the generator `Accessor` as an example, 
+but the things explained in this tutorial apply to every other generator.
+
+You use Java::Geci by using the `Geci` class in tests.
+
+```java
+public class TestAccessor {
+
+    @Test
+    public void testAccessor() throws Exception {
+        Assertions.assertFalse(new Geci().source(maven()
+                        .module("javageci-examples").mainSource())
+                        .register(Accessor.builder().build()).generate(),
+                Geci.FAILED);
+    }
+}
+```
+This might seem like a lot at first, so let's break it down.
+
+`Geci` has a fluent interface.
+
+The method `source()` can be used to specify the directories where your
+source files are. If you have source files in different places you
+have to chain several `source()` invocations one after the other. Every
+single call to `source()` can specify several directories. These count
+as one single directory and the first that exists is used to discover
+the files.<br/>
+This means:
+* Call `source("foo").source("bar")` if you have sources in **both** foo **and** bar.
+* Call `source("foo", "bar")` if you have sources in **either** foo **or** bar.
+*maven().module("javageci-examples").mainSource()* is just "javageci-examples/src"
+
+The method `register()` can register one or more source code generators. 
+You can also chain `register()` calls and register your generators one-by-one. 
+It's up to you. Each registered generator will be invoked on the sources.
+ 
+Finally the method invocation `generate()` will do the work, read the
+source files and generate the code.
+
+Every generator has a mnemonic which identifies the generator. For example the Accessor
+generator (that generates getters and setters) has the mnemonic "accessor":
+
+```java
+public String mnemonic() { return "accessor" }
+```
+
+After you assigned the source to the generator this mnemonic is used to identify 
+which classes need code generation. This can be done in two ways:
+
+* Adding the mnemonic of the generator in a @Geci annotation. <br/>
+Example (using the Accessor generator):
+
+```java
+@Geci("accessor")
```

Comment:
You can also have the line commented out in case you do not want to have a dependency on any annotation during run-time. So

```
//@Geci("accessor")
public class Example {
```

also works. But it is not recommended. Better just use the segment editor-fold with the id.

Reply:
This is probably out-of-scope for this tutorial just like creating your own annotations.

Reply:
Added a reference to another tutorial about writing your own annotations for Java::Geci, will be added there.